src/webapi/custom_elements.rs

```diff
@@ -0,0 +1,62 @@
+use webcore::value::Reference;
+webapi::error::TypeError;
+
+use webapi::dom_exception::{
+    NotSupportedError,
+    SyntaxError
+};
+
+#[cfg(feature = "experimental_features_which_may_break_on_minor_version_bumps")]
+use webcore::promise::{Promise, TypedPromise};
+
+/// The CustomElementRegistry interface provides methods for registering custom elements 
+/// and querying registered elements. To get an instance of it, use the window.customElements property. 
+///
+/// [(JavaScript docs)](hhttps://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry)
+/// https://html.spec.whatwg.org/multipage/custom-elements.html#customelementregistry
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "CustomElementRegistry")]
+pub struct CustomElementRegistry(Reference);
+
+
+error_enum_boilerplate! {
+    /// A enum of the exceptions that CustomElementRegistry.define() may throw
+    DefineError,
```

Comment:
A little more verbose name would be better here, I think. Maybe `CustomElementDefineError`? It's a little long, but simply `DefineError` seems to generic.